### DISCUSSION TITLE
2021-46: GPG 增加新的 Subkey

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#284] [Announcements]
# 2021-46: GPG 增加新的 Subkey

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2021-46/

**Comment by movsb** [comment]
还不太熟悉 gpg 的日常使用，想请教一下：为什么 delkey 等命令不需要像 addkey 那样需要用 passphrase 解锁 primary keypair 才能操作？

**Reply by Xuanwo** [reply]
我在本地试了一下确实如此：创建一个新的带 passphrase 的 keypair

- `--edit-key` 进入编辑模式不需要 passphrase
- `delkey` 不需要 passphrase
- `addkey` 需要 primary secert key 的 passphrase

我查了一下 [GPG 的文档](https://www.gnupg.org/gph/en/manual/c235.html)，有一处提到了一些：

> When a subkey or user ID is generated it is self-signed using your master signing key, which is why you must supply your passphrase when the key is generated.

反之，`delkey` 并不需要执行 `signing key` 操作，我想这就是原因吧。

**Reply by movsb** [reply]
嗯，这个可以理解。

另外一点是，如果删除不需要 passphrase 的话，别人如果用我电脑就可以任意删除我的 subkey 了。我的 primary keypair 和 subkey 目前是放在一起的（可能是不正确的日常使用方法）。

**Reply by Xuanwo** [reply]
电脑被物理接触的话，很多安全假设就彻底消失啦。还是取决你平时场景和实际需求，如果是日常放在家里的电脑，那现在这样就挺好；如果要带到办公场所或者是经常带出去的移动设备，我觉得把 primary secret key 删掉并且做好备份很有必要，或者可以考虑买一个 Yubico 之类的物理密钥～

**Reply by movsb** [reply]
懂了，感谢指点 😊
